### PR TITLE
Add a new forceOAuthLogin option to appSlice

### PR DESCRIPTION
**Pull request by VictorVelarde**
`forceOAuthLogin`: if configured via appSlice, a new fullscreen Login protects the whole app

image

This is compatible with the (current) presence of a UserMenu, to Logout or Go to Dashboard.

### CODE REVIEW

File: template-sample-app/template/src/components/views/login/Login.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      dispatch(setError(`OAuth error: ${oauthParams.error}`));
+    } else {
+      dispatch(setTokenAndUserInfoAsync(oauthParams));
+    }
+  };
+
+  const [handleLogin] = useOAuthLogin(oauthApp, onParamsRefreshed);
+
+  return (
+    <Container className={classes.containerWrapper}>
```
Comment (neokore) [inline template-sample-app/template/src/components/views/login/Login.js:37]: Maybe a Grid component here is better, unless you've found any issue to use it here

File: template-skeleton/template/src/components/views/login/Login.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      dispatch(setError(`OAuth error: ${oauthParams.error}`));
+    } else {
+      dispatch(setTokenAndUserInfoAsync(oauthParams));
+    }
+  };
+
+  const [handleLogin] = useOAuthLogin(oauthApp, onParamsRefreshed);
+
+  return (
+    <Container className={classes.containerWrapper}>
```
Comment (neokore) [inline template-skeleton/template/src/components/views/login/Login.js:44]: Same as sample-app

**Review by neokore — APPROVED** [review]
Haven't tested live, but looks good!

File: template-sample-app/template/src/components/views/login/Login.js (hunk 1 above)
Comment (VictorVelarde) [reply template-sample-app/template/src/components/views/login/Login.js:37]: Merging now, but I'll consider it when we apply the new design, thx!

**Pull request merged**